flexget/components/managed_lists/lists/ombi_list.py

```diff
@@ -0,0 +1,616 @@
+            return None
+
+        data = response.json()
+
+        # Their is a bug in OMBI where if you get a movie by its TMDB ID
+        # then the theMovieDbId field will be blank for some reason...
+        data['theMovieDbId'] = tmdb_id
+
+        return OmbiTv(base_url, auth, data)
+
+
+class OmbiSet(MutableSet):
+    """The schema for the Ombi managed list."""
+
+    supported_ids = ['imdb_id', 'tmdb_id', 'ombi_id']
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'api_key': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'type': {'type': 'string', 'enum': ['shows', 'seasons', 'episodes', 'movies']},
+            'only_approved': {'type': 'boolean', 'default': True},
+            'include_available': {'type': 'boolean', 'default': False},
+            'include_year': {'type': 'boolean', 'default': False},
+            'include_ep_title': {'type': 'boolean', 'default': False},
+        },
+        'oneOf': [{'required': ['username', 'password']}, {'required': ['api_key']}],
+        'required': ['url', 'type'],
+        'additionalProperties': False,
+    }
+
+    @property
+    def immutable(self):
+        return False
+
+    def __init__(self, config: Dict[str, Any]):
+        self.config = config
+        self._items = None
+        self._cached_items = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def add(self, entry: Entry):
+        log.info(f"Adding {entry['title']} to ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+
+        log.info(f"{entry['title']} was added to ombi_list.")
+
+    def __ior__(self, entries: List[Entry]):
+        for entry in entries:
+            self.add(entry)
+
+    def discard(self, entry: Entry):
+        """Removes an entry from OMBI by marking it as available.
+
+        Args:
+            entry (Entry): An item from a task.
+        """
+
+        log.info(f"Removing {entry['title']} from ombi_list.")
+
+        log.debug(f"Getting OMBI entry for {entry['title']}.")
+
+        ombi_entry = self._get_ombi_entry(entry)
+
+        if not ombi_entry:
+            log.error(f"Failed to find OMBI entry for {entry['title']}.")
+            return
+
+        ombi_entry.mark_requested()
+        ombi_entry.mark_available()
+
+        log.info(f"{entry['title']} was removed from ombi_list.")
+
+    def __isub__(self, entries: List[Entry]):
+        for entry in entries:
+            self.discard(entry)
+
+    def _find_entry(self, entry: Entry):
+        if "imdb_id" not in entry:
```

Comment:
Need to test if imdb_id is None as well.. This test will check if there is an entry named imdb_Id but not if it contains anything in my testing it died here with empty IDs.

Reply:
I will work on this soon, but I will likely not work on it over the weekend.

Reply:
> I will work on this soon, but I will likely not work on it over the weekend.

If not you may see a PR from me  assuming my fork and branches are working now.

Reply:
If you look at the tvdb_list, they loop over the supported IDs checking if any are valid. That was the solution I was going to try out.

Reply:
Pulled the latest version which has a lot of changes in it.. When I loaded it up it would not iterate through my ombi movie list at all anymore.. Had a little time to look at it but I think the issue is that moving to the v2/movie call isn't working because it has several required parameters. I will see if I can poke at it more over the weekend

Reply:
@bagobones I have pushed a commit that modifies `_find_entry` to use all the supported ID's when searching. I'm on vacation and unable to test but it should have fixed your issue. 

I tested all my previous work but only by adding and removing from the list. I haven't done any testing with matching. Can you provide an example of what kind of testing you are doing?

Reply:
@shadycuz 

Nothing wild just dumping out my VERY long request list but it contains a lot of items missing IDs etc

```
  ombi_list_all:
    disable: seen
    ombi_list:
      url: '{? ombi.url ?}'
      api_key: '{? ombi.api_key ?}'
      type: movies
    accept_all: yes
    exec: echo "Found T:{{title}} request:{{ombi_request_id}} TMDB:{{tmdb_id}} imdbId:{{imdb_id}}"
```

The commit before last ran, but would just skip over the bad entries with some errors. Now the plugin crashes.

I might try and setup a TEST ombi instance that contains clean entries and start from there to see if it runs. I have had a number of other projects to get done this weekend however so it isn't looking like I will spend much time coding..

Reply:
These optional fields without a Jinja fallback give me goosebumps

Reply:
@bagobones I did indeed make a mistake. I was using GitHub Co-pilot (ChatGPT) and when I did that refactoring it changed `/api/v1/Request/movie` to `/api/v2/Requests/movie` without me catching it.

I have fixed that and found some other bugs with how items from Ombi were being fetched and filtered. By default the list will only show `approved` requests but you can change that might configuring the `status` field to something different. If you want all requests use `all` but that will also include denied. I don't think people will use `all` that often, typically you will want a certain type, like `approved` or `requested`, where requested is not approved or denied (yet). 

Give it another shot and let me know how it goes.

Reply:
Spent most of the weekend doing house projects.. Didn't get to take a look.

ON the topic of GitHub Co-pilot (ChatGPT), I find it works best as an aid for code structure, hints how to write things or if you ask it what is wrong it actually has some good answers if you feed it the errors.. Asking it to write API code for things is was not explicitly trained on will cause it to make up (hallucinate) solutions.

Also checking in untested code is a major programing sin. LOL

At any rate I will test again when I get a chance, on weekdays it it's usually means late evenings, if I can contribute fixes I will but it seems as long as I comment on the problem you have it fixed by the time I am off work the next day.

I will try and work my way through testing all of the options you have added. At this point it looks like it SHOULD cover all the use cases I can think of and if it all works be a robust option for people to use.

Reply:
I think the hardest thing to test will be TV. I don't really understand stand how flexget separates and tracks things like TV Show vs Season vs Episode. 

So I'm not sure if you request one in Ombi, that you will get the desired result in flexget. I'm also not sure on how to even test it.

Reply:
> I think the hardest thing to test will be TV. I don't really understand stand how flexget separates and tracks things like TV Show vs Season vs Episode.
> 
> So I'm not sure if you request one in Ombi, that you will get the desired result in flexget. I'm also not sure on how to even test it.

So the list can be used directly with discover or with built in flexget plugins.

IIRC, if you where to use this with the series  / configure series plugin, it would only care about info it found and what ITs configuration was, IE I think it would only work with the show, and episode modes in terms of granularity where it would take the series name or the episodes if passed them.

However if you use the list with SITE plugins, the output of title, title S02, title S02E04 for each of the types can in theory be used as a search string, allowing you to get different results.

@gazpachoking  might be best to chime in on this in case I am wrong.